.github/workflows/test-enos-scenario-ui.yml

```diff
@@ -0,0 +1,141 @@
+      ENOS_VAR_terraform_plugin_cache_dir: ./support/terraform-plugin-cache
+      ENOS_VAR_vault_license_path: ./support/vault.hclic
+      GOPRIVATE: github.com/hashicorp
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set Up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ needs.get-metadata.outputs.go-version }}
+      - uses: hashicorp/action-setup-enos@v1
+        with:
+          github-token: ${{ secrets.ELEVATED_GITHUB_TOKEN }}
+      - name: Set Up Git
+        run: git config --global url."https://${{ secrets.elevated_github_token }}:@github.com".insteadOf "https://github.com"
+      - name: Set Up Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ needs.get-metadata.outputs.node-version }}
+      - name: Set Up Terraform
+        uses: hashicorp/setup-terraform@v2
+        with:
+          cli_config_credentials_token: ${{ secrets.TF_API_TOKEN }}
+          terraform_wrapper: false
+      - name: Prepare scenario dependencies
+        run: |
+          mkdir -p ./enos/support/terraform-plugin-cache
+          echo "${{ secrets.SSH_KEY_PRIVATE_CI }}" > ./enos/support/private_key.pem
+          chmod 600 ./enos/support/private_key.pem
+      - name: Set Up Vault Enterprise License
+        if: contains(${{ github.event.repository.name }}, 'ent')
+        run: echo "${{ secrets.VAULT_LICENSE }}" > ./enos/support/vault.hclic || true
+      - name: Set Up Chrome
+        if: contains(${{ github.event.repository.name }}, 'ent')
+        run: |
+          wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
+          sudo dpkg -i google-chrome-stable_current_amd64.deb
+      - name: Configure AWS credentials from Test account
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID_CI }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY_CI }}
+          aws-region: us-east-1
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN_CI }}
+          role-skip-session-tagging: true
+          role-duration-seconds: 3600
+      - name: Setup Cluster
+        id: setup_cluster
+        env:
+          ENOS_VAR_ui_run_tests: false
+        # Continue once and retry to handle occasional blips when creating infrastructure.
+        continue-on-error: true
+        run: enos scenario launch --timeout 60m0s --chdir ./enos ui edition:${{ needs.get-metadata.outputs.vault_edition }} backend:${{ inputs.storage_backend }}
+      - name: Retry Setup Cluster
+        id: setup_cluster_retry
+        if: steps.setup_cluster.outcome == 'failure'
+        env:
+          ENOS_VAR_ui_run_tests: false
+        run: enos scenario launch --timeout 60m0s --chdir ./enos ui edition:${{ needs.get-metadata.outputs.vault_edition }} backend:${{ inputs.storage_backend }}
+      - name: Run UI Tests
```

Comment:
Yeah, I like this approach. For this scenario it was necessary though, since if the only reason the scenario fails is due to the UI tests failing, we would not want to re-run it.